usecase/src/main/java/org/mobilitydata/gtfsvalidator/usecase/ValidateNoOverlappingStopTimeInTripBlock.java

```diff
@@ -0,0 +1,364 @@
+                        otherTripId,
+                        timeUtils.convertIntegerToHMMSS(otherTripFirstTime),
+                        timeUtils.convertIntegerToHMMSS(otherTripLastTime),
+                        blockId,
+                        tripCalendar.getOverlappingDays(unvisitedTripCalendar)));
+    }
+
+    /**
+     * Utility method to determine if trips are overlapping when trips do not have the same service_id and file
+     * `calendar.txt` is not provided in the GTFS archive.
+     *
+     * @param blockId                id of the block
+     * @param trip                   first {@link Trip} to analyze
+     * @param tripId                 id of {@link Trip}
+     * @param otherTrip              other {@link Trip} to analyze
+     * @param otherTripId            other {@link Trip} id
+     * @param tripFirstTime          {@link Trip} first time
+     * @param tripLastTime           {@link Trip} last time
+     * @param otherTripFirstTime     other {@link Trip} first time
+     * @param otherTripLastTime      other {@link Trip} last time
+     * @param otherTripFirstStopTime other {@link Trip} first {@link StopTime}
+     * @param otherTripLastStopTime  other {@link Trip} last {@link StopTime}
+     */
+    private void checkOverlappingTripsDifferentServiceIdCalendarNotProvided(final String blockId,
+                                                                            final Trip trip,
+                                                                            final String tripId,
+                                                                            final Trip otherTrip,
+                                                                            final String otherTripId,
+                                                                            final int tripFirstTime,
+                                                                            final int tripLastTime,
+                                                                            final Integer otherTripFirstTime,
+                                                                            final Integer otherTripLastTime,
+                                                                            final StopTime otherTripFirstStopTime,
+                                                                            final StopTime otherTripLastStopTime) {
+        // interpret data from `calendar_dates.txt` to determine if currentTrip and
+        // otherTrip operate on same days
+        final Map<String, CalendarDate> tripCalendarDateCollection =
+                dataRepo.getCalendarDateAll().get(trip.getServiceId());
+        final Map<String, CalendarDate> unvisitedTripCalendarDateCollection =
+                dataRepo.getCalendarDateAll().get(otherTrip.getServiceId());
+        final Set<String> currentTripServiceDates = new HashSet<>();
+        tripCalendarDateCollection.forEach((date, calendarDate) -> {
```

Comment:
I believe this should be:

```java
        tripCalendarDateCollection.forEach((serviceId, calendarDate) -> {
```

...which means that the below line is adding the service ID and not the LocalDate to the HashSet, and needs to be:

```java
                currentTripServiceDates.add(calendarDate);
```

Reply:
Oh, looking at `InMemoryGtfsDataRepository`, this string is actually a string representation of the date. Which is kind of confusing here, although I understand why we did it (to check for duplicates when importing calendar_dates.txt).

I think it's safer to add the `LocalDate` to the set here (i.e., the `calendarDate` field) as it's not a (somewhat) arbitrary string key based on the repository implementation and is more specific to our domain of date checking in this method.

Reply:
Actually, in the in memory repo, could we instead do:

```java
Map<String, Set<CalendarDate>> calendarDatePerServiceIdAndDate = new HashMap<>();
```

For data like the Netherlands with all service in calendar_dates.txt it would save a ton of duplicate strings in memory. Although we'd need to implement hashcode() and equals() for CalendarDate. Maybe open a new issue for this change?